File: 14_longest_common_prefix/longestCommonPrefix2.ts
function longestCommonPrefix(strs: string[]): string {
  const n = Math.min(...strs.map((i) => i.length));
  let res = "";

  for (let i = 0; i < n; i++) {
    const curr = strs[0][i];

    for (let s of strs) {
      if (s[i] !== curr) return res;
    }

    res += curr;
  }

  return res;
}

File: 14_longest_common_prefix/longestCommonPrefix2.go
func longestCommonPrefix(strs []string) string {
	if len(strs) == 0 {
		return ""
	}

	var res strings.Builder

	for i := 0; i < len(strs[0]); i++ {
		currentChar := strs[0][i]
		for _, s := range strs[1:] {
			if i >= len(s) || s[i] != currentChar {
				return res.String()
			}
		}
		res.WriteByte(currentChar)
	}

	return res.String()
}